src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -2675,6 +2675,135 @@ static void Main(string[] args)
+        // be collection expressions, then MethodTypeInferrer.MakeOutputTypeInferences
+        // will probably need to make an output type inference from each element within
+        // the nested spread element collection expression.
+        [Fact]
+        public void TypeInference_Spread_04()
+        {
+            string source = """
+                using System;
+                class Program
+                {
+                    static Func<T>[] F<T>(Func<T>[] arg) => arg;
+                    static void Main()
+                    {
+                        var x = F([null, .. [() => 1]]);
+                        x.Report();
+                    }
+                }
+                """;
+            var comp = CreateCompilation(new[] { source, s_collectionExtensions });
+            comp.VerifyEmitDiagnostics(
+                // 0.cs(7,29): error CS9176: There is no target type for the collection expression.
+                //         var x = F([null, .. [() => 1]]);
+                Diagnostic(ErrorCode.ERR_CollectionExpressionNoTargetType, "[() => 1]").WithLocation(7, 29));
+        }
+
+        [Fact]
+        public void TypeInference_Spread_05()
+        {
+            string source = """
+                class Program
+                {
+                    static T[] F<T>(T[] arg) => arg;
+                    static void Main()
+                    {
+                        object x = new[] { 2, 3 };
+                        var y = F([..x]);
+                        var z = F([1, ..x]);
+                    }
+                }
+                """;
+            var comp = CreateCompilation(new[] { source, s_collectionExtensions });
+            comp.VerifyEmitDiagnostics(
+                // 0.cs(7,22): error CS1579: foreach statement cannot operate on variables of type 'object' because 'object' does not contain a public instance or extension definition for 'GetEnumerator'
+                //         var y = F([..x]);
+                Diagnostic(ErrorCode.ERR_ForEachMissingMember, "x").WithArguments("object", "GetEnumerator").WithLocation(7, 22),
+                // 0.cs(8,25): error CS1579: foreach statement cannot operate on variables of type 'object' because 'object' does not contain a public instance or extension definition for 'GetEnumerator'
+                //         var z = F([1, ..x]);
+                Diagnostic(ErrorCode.ERR_ForEachMissingMember, "x").WithArguments("object", "GetEnumerator").WithLocation(8, 25));
+        }
+
+        [Fact]
+        public void TypeInference_Spread_06()
+        {
+            string source = """
+                class Program
+                {
+                    static T[] F<T>(T[] arg) => arg;
+                    static void Main()
+                    {
+                        dynamic x = new[] { 2, 3 };
+                        var y = F([..x]);
+                        y.Report(includeType: true);
+                        var z = F([1, ..x]);
```

Comment:
nit: might be nice to show a case with multiple spread arguments.

Reply:
Added to `TypeInference_Spread_01`.